pkg/frontend/quota_validation.go

```diff
@@ -0,0 +1,178 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+)
+
+const (
+	standardDSv3    = "standardDSv3Family"
+	standardDASv4   = "standardDASv4Family"
+	standardESv3    = "standardESv3Family"
+	standardEISv4   = "standardEISv4Family"
+	standardEIDSv4  = "standardEIDSv4Family"
+	standardEIv5    = "standardEIv5Family"
+	standardEISv5   = "standardEISv5Family"
+	standardEIDSv5  = "standardEIDSv5Family"
+	standardEIDv5   = "standardEIDv5Family"
+	standardFSv2    = "standardFSv2Family"
+	standardMS      = "standardMSFamily"
+	standardGFamily = "standardGFamily"
+	standardLSv2    = "standardLsv2Family"
+	standardNCAS    = "Standard NCASv3_T4 Family"
+)
+
+type QuotaValidator interface {
+	ValidateQuota(ctx context.Context, azEnv *azureclient.AROEnvironment, environment env.Interface, subscriptionID, tenantID string, oc *api.OpenShiftCluster) error
+	//ValidateNewCluster(ctx context.Context, envValue env.Interface, subscription *api.SubscriptionDocument, cluster *api.OpenShiftCluster, staticValidator api.OpenShiftClusterStaticValidator, ext interface{}, path string) error
+}
+
+type quotaValidator struct{}
+
+func addRequiredResources(requiredResources map[string]int, vmSize api.VMSize, count int) error {
+	vmTypesMap := map[api.VMSize]struct {
```

Comment:
We keep a similar list in pkg/api/validate/vm.go, would it be possible to combine them so we only need to update one list when we add sizes?